File: src/examples/local-api/snippets.ts
// read the discovery file (Windows path shown)
import { readFile } from "node:fs/promises";

const { port, token } = JSON.parse(
  await readFile(`${process.env.APPDATA}/viewbus/local-api.json`, "utf8"),
);
const base = `http://127.0.0.1:${port}`;
const headers = { Authorization: `Bearer ${token}` };
//---
const status = await fetch(`${base}/v1/status`, { headers }).then((r) => r.json());
//---
const { results } = await fetch(
  `${base}/v1/search?q=orders&limit=10`,
  { headers },
).then((r) => r.json());
//---
await fetch(`${base}/v1/focus`, {
  method: "POST",
  headers: { ...headers, "Content-Type": "application/json" },
  body: JSON.stringify({ id: results[0].id }),
});

File: src/examples/local-api/snippets.py
# read the discovery file (Windows path shown)
import json, os, requests

info = json.load(open(os.path.join(os.environ["APPDATA"], "viewbus", "local-api.json")))
base = f"http://127.0.0.1:{info['port']}"
headers = {"Authorization": f"Bearer {info['token']}"}
#---
status = requests.get(f"{base}/v1/status", headers=headers).json()
#---
results = requests.get(
    f"{base}/v1/search",
    headers=headers,
    params={"q": "orders", "limit": 10},
).json()["results"]
#---
requests.post(f"{base}/v1/focus", headers=headers, json={"id": results[0]["id"]})
